### PR TITLE
[IMP] Migrated website_event_sale into new API

### PR DESCRIPTION
**Pull request by rga-odoo**
Task : https://www.odoo.com/web#id=12027&view_type=form&model=project.task&menu_id=3942&action=327
Pad: https://pad.odoo.com/p/r.4Tb7FCYAzt8iqyQi

### CODE REVIEW

File: addons/website_event_sale/models/sale_order.py [hunk 1]
```diff
@@ -1,40 +1,39 @@
...
-            if context.get("event_ticket_id"):
-                domain += [('event_ticket_id', '=', context.get("event_ticket_id"))]
-            return self.pool.get('sale.order.line').search(cr, SUPERUSER_ID, domain, context=context)
-
-    def _website_product_id_change(self, cr, uid, ids, order_id, product_id, qty=0, line_id=None, context=None):
-        values = super(sale_order, self)._website_product_id_change(cr, uid, ids, order_id, product_id, qty=qty, line_id=line_id, context=None)
+        domain = [('id', 'in', line_ids)]
+        for so in self:
+            if self._context.get("event_ticket_id"):
+                domain += [('event_ticket_id', '=', self._context.get("event_ticket_id"))]
```
Comment (xmo-odoo) [inline addons/website_event_sale/models/sale_order.py]: Why use `extend`/+= here instead of `append`?

File: addons/website_event_sale/models/sale_order.py [hunk 2]
```diff
@@ -1,40 +1,39 @@
...
-        for so in self.browse(cr, uid, ids, context=context):
-            domain = [('id', 'in', line_ids)]
-            if context.get("event_ticket_id"):
-                domain += [('event_ticket_id', '=', context.get("event_ticket_id"))]
-            return self.pool.get('sale.order.line').search(cr, SUPERUSER_ID, domain, context=context)
-
-    def _website_product_id_change(self, cr, uid, ids, order_id, product_id, qty=0, line_id=None, context=None):
-        values = super(sale_order, self)._website_product_id_change(cr, uid, ids, order_id, product_id, qty=qty, line_id=line_id, context=None)
+        domain = [('id', 'in', line_ids)]
+        for so in self:
```
Comment (xmo-odoo) [inline addons/website_event_sale/models/sale_order.py]: What is the point of the iteration? `so` is not used internally, we're just getting a value from the context.

File: addons/website_event_sale/models/sale_order.py [hunk 3]
```diff
@@ -1,40 +1,39 @@
...
+            if self._context.get("event_ticket_id"):
```
Comment (xmo-odoo) [inline addons/website_event_sale/models/sale_order.py]: I would suggest `self.env.context`, it's slightly longer but probably more sensible.

File: addons/website_event_sale/models/sale_order.py [hunk 4]
```diff
@@ -1,40 +1,39 @@
...
+                domain += [('event_ticket_id', '=', self._context.get("event_ticket_id"))]
+            return self.env['sale.order.line'].sudo().search(domain).ids
 
+    @api.multi
+    def _website_product_id_change(self, order_id, product_id, qty=0, line_id=None):
+        values = super(SaleOrder, self)._website_product_id_change(order_id, product_id, qty=qty, line_id=line_id)
         event_ticket_id = None
-        if context.get("event_ticket_id"):
-            event_ticket_id = context.get("event_ticket_id")
+        if self._context.get("event_ticket_id"):
```
Comment (xmo-odoo) [inline addons/website_event_sale/models/sale_order.py]: `self.env.context`

File: addons/website_event_sale/models/sale_order.py [hunk 5]
```diff
@@ -43,23 +42,19 @@ def _website_product_id_change(self, cr, uid, ids, order_id, product_id, qty=0,
...
-            line = OrderLine.browse(cr, uid, line_id, context=context)
+            line = OrderLine.browse(line_id)
             ticket = line.event_ticket_id
             old_qty = int(line.product_uom_qty)
         else:
-            line, ticket = None, None
-            ticket_ids = Ticket.search(cr, uid, [('product_id', '=', product_id)], limit=1, context=context)
-            if ticket_ids:
-                ticket = Ticket.browse(cr, uid, ticket_ids[0], context=context)
+            line, ticket = None, self.env['event.event.ticket'].search([('product_id', '=', product_id)], limit=1) or None
```
Comment (xmo-odoo) [inline addons/website_event_sale/models/sale_order.py]: split that over two separate assignments for clarity

File: addons/website_event_sale/models/sale_order.py [hunk 6]
```diff
@@ -78,21 +73,20 @@ def _cart_update(self, cr, uid, ids, product_id=None, line_id=None, add_qty=0, s
...
                     ('state', '!=', 'cancel'),
-                    ('sale_order_id', '=', ids[0])
+                    ('sale_order_id', '=', self.ids[0])
                     ('event_ticket_id', '=', ticket.id)
-                ], offset=new_qty, limit=(old_qty-new_qty),
-                order='create_date asc', context=context)
-            Attendee.button_reg_cancel(cr, uid, attendees, context=context)
+                ], offset=new_qty, limit=(old_qty - new_qty),
+                order='create_date asc')
+            Attendee.button_reg_cancel(attendees)
```
Comment (xmo-odoo) [inline addons/website_event_sale/models/sale_order.py]: Why is that not `attendees.button_reg_cancel`? And then `Attendee` could be inlined.

File: addons/website_event_sale/models/sale_order.py [hunk 7]
```diff
@@ -78,21 +73,20 @@ def _cart_update(self, cr, uid, ids, product_id=None, line_id=None, add_qty=0, s
...
 
-        values.update(super(sale_order, self)._cart_update(
-            cr, uid, ids, product_id, line_id, add_qty, set_qty, context, **kwargs))
+        values.update(super(SaleOrder, self)._cart_update(product_id, line_id, add_qty, set_qty, **kwargs))
 
         # removing attendees
         if ticket and new_qty < old_qty:
             attendees = Attendee.search(
-                cr, uid, [
+                [
```
Comment (xmo-odoo) [inline addons/website_event_sale/models/sale_order.py]: Not really necessary, but this bracket opening could probably move up to the next line

File: addons/website_event_sale/models/sale_order.py (hunk 2 above)
Comment (tde-banana-odoo) [inline addons/website_event_sale/models/sale_order.py]: This is an example of "do not migrate line by line". This code was obviously wrong and badly implemented, please simplify and clean the whole method here.

File: addons/website_event_sale/models/sale_order.py (hunk 5 above)
Comment (tde-banana-odoo) [inline addons/website_event_sale/models/sale_order.py]: Anyway this kind of code is unreadable. Keep it in several lines.

File: addons/website_event_sale/static/src/js/website.tour.event_sale.js [hunk 8]
```diff
@@ -31,7 +31,7 @@
             {
                 title:     "Fill attendees details",
                 waitFor:   'form[id="attendee_registration"] .btn:contains("Continue")',
-                autoComplete: function (tour) {
+                autoComplete: function () {
```
Comment (tde-banana-odoo) [inline addons/website_event_sale/static/src/js/website.tour.event_sale.js]: Why this change ? This has probably nothing to do with this branch.

File: addons/website_event_sale/static/src/js/website.tour.event_sale.js [hunk 9]
```diff
@@ -65,7 +65,7 @@
             {
                 title:     "Complete the checkout",
                 element:   'form[action="/shop/confirm_order"] .btn:contains("Confirm")',
-                onload: function (tour) {
+                onload: function () {
```
Comment (tde-banana-odoo) [inline addons/website_event_sale/static/src/js/website.tour.event_sale.js]: Why this change ? This has probably nothing to do with this branch.

File: addons/website_event_sale/models/sale_order.py (hunk 1 above)
Comment (dja-odoo) [inline addons/website_event_sale/models/sale_order.py]: Done.

File: addons/website_event_sale/models/sale_order.py (hunk 2 above)
Comment (dja-odoo) [inline addons/website_event_sale/models/sale_order.py]: Right. I removed the iteration.

File: addons/website_event_sale/models/sale_order.py (hunk 3 above)
Comment (dja-odoo) [inline addons/website_event_sale/models/sale_order.py]: Done!

File: addons/website_event_sale/models/sale_order.py (hunk 5 above)
Comment (dja-odoo) [inline addons/website_event_sale/models/sale_order.py]: Done.

File: addons/website_event_sale/models/sale_order.py (hunk 6 above)
Comment (dja-odoo) [inline addons/website_event_sale/models/sale_order.py]: Done.

File: addons/website_event_sale/models/sale_order.py (hunk 7 above)
Comment (dja-odoo) [inline addons/website_event_sale/models/sale_order.py]: Done.

File: addons/website_event_sale/static/src/js/website.tour.event_sale.js (hunk 8 above)
Comment (dja-odoo) [inline addons/website_event_sale/static/src/js/website.tour.event_sale.js]: :scream: Yep!! Sorry for the mess. Removed it.

File: addons/website_event_sale/static/src/js/website.tour.event_sale.js (hunk 9 above)
Comment (dja-odoo) [inline addons/website_event_sale/static/src/js/website.tour.event_sale.js]: Done.

File: addons/website_event_sale/models/sale_order.py [hunk 10]
```diff
@@ -78,21 +72,19 @@ def _cart_update(self, cr, uid, ids, product_id=None, line_id=None, add_qty=0, s
...
         if ticket and new_qty < old_qty:
-            attendees = Attendee.search(
-                cr, uid, [
+            attendees = self.env['event.registration'].search([
                     ('state', '!=', 'cancel'),
-                    ('sale_order_id', '=', ids[0])
-                    ('event_ticket_id', '=', ticket.id)
+                    ('sale_order_id', '=', self.ids[0]),
+                    ('event_ticket_id', '=', ticket.id),
                 ], offset=new_qty, limit=(old_qty-new_qty),
```
Comment (rga-odoo) [inline addons/website_event_sale/models/sale_order.py]: stdin:83:50: E226 missing whitespace around arithmetic operator

Comment (dja-odoo) [inline addons/website_event_sale/models/sale_order.py]: Done.

**Comment by msh-odoo** [comment]
Why try except in _add_event, shouldn't we pass raise_if_not_found

**Comment by jem-odoo** [comment]
Merged at 6e28d2071db585873a5bd6e6081f37628eefa599
